mooncake-store/src/master_service.cpp

```diff
@@ -200,6 +200,19 @@ ErrorCode MasterService::GetReplicaList(
     return ErrorCode::OK;
 }
 
+ErrorCode MasterService::BatchGetReplicaList(
+    const std::vector<std::string>& keys,
+    std::unordered_map<std::string, std::vector<Replica::Descriptor>>&
+        batch_replica_list) {
+    for (const auto& key : keys) {
```

Comment:
I'm not sure putting the batch processing directly in the master is the best approach, as it might lose some parallelism.

I think a better option could be to put it in the master client instead. We could give each request its own task there, which would allow for better parallel execution. What do you think of that idea?